### PR TITLE
Feature/feature personal notification

### PR DESCRIPTION
**Pull request by enrichp**

### CODE REVIEW

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Event.java  18
         

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrNewPersonalEvent.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeletePersonalEvent.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeletePersonalEvent.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrNewPersonalEvent.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5302497&bid=17354832)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/calendar/CalendarFragment.java [hunk 1]
```diff
@@ -18,52 +35,309 @@
...
+    /**
+     * Initialize the dialog elements.
+     * @param timeText The time
+     * @param sp The spinner
+     * @param reasonText The event description
+     * @param dateText The event date
+     * @param layout The layout of the dialog
+     * @param newPersonal The dialog
+     */
+    private void dialogElements(MaterialAlertDialogBuilder newPersonal, LinearLayout layout, EditText reasonText,
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/calendar/CalendarFragment.java:109]: Codacy Issue found: [More than 5 parameters (found 6).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5302497)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/calendar/CalendarFragment.java [hunk 2]
```diff
@@ -18,52 +35,309 @@
...
+                selectedPet = pet;
+            }
+        }
+    }
+
+    /**
+     * Gets all the pets of the user.
+     * @return An array with the pets name
+     */
+    private ArrayList<?> getPetsName() {
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/calendar/CalendarFragment.java:282]: Codacy Issue found: [Avoid using implementation types like 'ArrayList'; use the interface instead](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5302497)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/calendar/CalendarFragment.java [hunk 3]
```diff
@@ -18,52 +35,309 @@
...
+
+    /**
+     * Indicates is a string is in a correct time format.
+     * @param time The time string
+     * @return True if the string is in a correct time format, false otherwise
+     */
+    private boolean isValidTime(String time) {
+        boolean result = false;
+        if (time.length() == TIME_LENGTH && time.indexOf(':') == POS2 && time.lastIndexOf(':') == POS5) {
+            if (Character.isDigit(time.charAt(0)) && Character.isDigit(time.charAt(1))
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/calendar/CalendarFragment.java:253]: Codacy Issue found: [These nested if statements could be combined](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5302497)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/calendar/CalendarFragment.java [hunk 4]
```diff
@@ -18,52 +35,309 @@
...
+     * Create a new personal event.
+     * @param dateText The date of the event
+     * @param reasonText The reason of the event
+     * @param sp The spinner for pet selection
+     * @param timeText The hour of the event
+     */
+    private void createPersonalEvent(EditText reasonText, TextView dateText, EditText timeText, Spinner sp) {
+        String petName = sp.getSelectedItem().toString();
+        StringBuilder dateTime = new StringBuilder(dateText.getText());
+        dateTime.append('T');
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/calendar/CalendarFragment.java:161]: Codacy Issue found: [StringBuffer (or StringBuilder).append is called consecutively without reusing the target variable.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5302497)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/calendar/CalendarFragment.java [hunk 5]
```diff
@@ -18,52 +35,309 @@
...
+     * Indicates is a string is in a correct time format.
+     * @param time The time string
+     * @return True if the string is in a correct time format, false otherwise
+     */
+    private boolean isValidTime(String time) {
+        boolean result = false;
+        if (time.length() == TIME_LENGTH && time.indexOf(':') == POS2 && time.lastIndexOf(':') == POS5) {
+            if (Character.isDigit(time.charAt(0)) && Character.isDigit(time.charAt(1))
+                    && Character.isDigit(time.charAt(POS3))) {
+                if (Character.isDigit(time.charAt(POS4)) && Character.isDigit(time.charAt(POS6))
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/calendar/CalendarFragment.java:255]: Codacy Issue found: [These nested if statements could be combined](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5302497)

**Review by xavier-campos-diaz — APPROVED** [review]

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrNewPersonalEvent.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrNewPersonalEvent.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeletePersonalEvent.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeletePersonalEvent.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5302497&bid=17354832)

File: app/src/main/res/values-ca/strings.xml [hunk 6]
```diff
@@ -89,6 +89,21 @@
     <string name="calendar_select_date">Selecciona una data</string>
     <string name="add_personal_event_button">Personal</string>
     <string name="add_periodic_event_button">Periòdic</string>
+    <string name="dialog_new_event">Nou event personal</string>
+    <string name="dialog_new_event_message">Introdueix la descripció de l\'event, l\'hora i selecciona la mascota que participarà.</string>
```
Comment (AlbertPG) [inline app/src/main/res/values-ca/strings.xml]: "Event" is incorrect in catalan. The translation from spanish would be "Esdeveniment".

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/views/CalendarEventsView.java [hunk 7]
```diff
@@ -35,10 +37,14 @@ public CalendarEventsView(Context context, @Nullable AttributeSet attrs) {
...
         List<Event> events = pet.getEvents(date);
-
         for (Event event : events) {
             PetComponentView petComponentView = new EventView(context, null, event).initializePetComponent(pet);
             addView(petComponentView);
+            this.petComponents.add(petComponentView);
         }
     }
+
+    public List<PetComponentView> getPetComponents() {
```
Comment (AlbertPG) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/views/CalendarEventsView.java:51]: You need to comment this method

**Review by AlbertPG — APPROVED** [review]
Comment the method and that's all folks

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Event.java  18
         

Complexity increasing per file
==============================
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrNewPersonalEvent.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeletePersonalEvent.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrNewPersonalEvent.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeletePersonalEvent.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5302497&bid=17354832)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Event.java  18
         

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrDeletePersonalEvent.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/TrNewPersonalEvent.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrNewPersonalEvent.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrDeletePersonalEvent.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5302497&bid=17354832)

**Pull request merged**